src/Assets/Api.php

```diff
@@ -88,26 +88,31 @@ public function register_script( $handle, $relative_src, $dependencies = [], $ha
 			$version = $this->get_file_version( $relative_src );
 		}
 
-		wp_register_script( $handle, $src, $dependencies, $version, true );
+		wp_register_script( $handle, $src, apply_filters( 'woocommerce_blocks_register_script_dependencies', $dependencies, $handle ), $version, true );
```

Comment:
I took this filter from `src/Assets.php`, we were inconsistently applying it in Assets but not in AssetsApi. Now that we have a unique way to register scripts, I think it makes sense to keep the filter.

Reply:
ahh that would explain it. The Payments API uses that to register the handle for payment method scripts as dependencies on certain handles (so various registration happens before the blocks load).